dataduct/utils/slack_hook.py

```diff
@@ -13,21 +13,33 @@ def post_message(message):
     Args:
         message(str): The message to post
     """
+    # If any of these fail, silently skip because the user doesn't know about
+    # the slack integration or doesn't care
     try:
         import slack
         import slack.chat
         config = Config()
         slack_config = config.etl['slack']
+    except KeyError:
+        return
+
+    # If any of these configs fail, output error message and fail because the
+    # user has misconfigured the slack integration
+    try:
         slack.api_token = slack_config['api_token']
         slack.chat.post_message(slack_config['channel_name'],
                                 message,
-                                username=slack_config['bot_username'])
-    except Exception:
-        logger.info('If you want to post a slack message when you activate a pipeline')  # noqa
-        logger.info('1) Run: pip install pyslack')
-        logger.info('2) Visit https://api.slack/com/web to generate a token')
-        logger.info('   and add:')
-        logger.info('       api_token:<your token>')
-        logger.info('       channel_name:<channel_name>')
-        logger.info('       bot_username:<bot_username>')
-        logger.info('   to the etl section of your config file')
+                                username=slack_config.get('bot_username',
+                                                          'Dataduct'))
+    except KeyError:
```

Comment:
you want to catch import error on slack as that might not be installed after the above change. 
So you may still want to catch a generic exception as posting on slack should never ever block you from running pipelines.